### PR TITLE
Add Python3 Support to ec2.py module

### PR DESCRIPTION
**Pull request by ZaiusDR**
##### ISSUE TYPE

 - Feature Pull Request

##### ANSIBLE VERSION

```
ansible 2.2.0.0
  config file =
  configured module search path = Default w/o overrides
```

##### SUMMARY

Role Task

```
- name: Create ec2 Test Instance
  ec2:
    region: "{{ region }}"
    zone: "{{ zone }}"
    keypair: "{{ keypair }}"
    instance_type: "{{ instance_type }}"
    image: "{{ ami }}"
    wait: yes
    group: "[{{ sg }}]"
    exact_count: "{{ exact_count }}"
    count_tag:
      Name: "{{ name_tag }}"
    instance_tags:
      Name: "{{ name_tag }}"
    volumes:
      - device_name: /dev/sda
        volume_size: 10
        delete_on_termination: true
  register: ec2
```

Errors while execution with python3:

```
TASK [create_ec2_instance : Create ec2 Test Instance] **************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'dict' object has no attribute 'iterkeys'
...
...
```

```
TASK [create_ec2_instance : Create ec2 Test Instance] **************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'dict' object has no attribute 'iteritems'
...
...
```

```
TASK [create_ec2_instance : Create ec2 Test Instance] **************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'function' object has no attribute 'func_code'
...
...
```

After changes it just works fine with both python2 and 3:

```
TASK [create_ec2_instance : Create ec2 Test Instance] **************************
changed: [localhost]
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @ZaiusDR for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by mscherer** [comment]
Seems ok for me, but I will let someone more knowledgeable with ec2 to verify too.

**Comment by gregdek** [comment]
Thanks @ZaiusDR for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by ZaiusDR** [comment]
Conflicts resolved and rebased!

**Comment by ZaiusDR** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @ZaiusDR for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/amazon/ec2.py [hunk 1]
```diff
@@ -614,7 +617,7 @@ def find_running_instances_by_count_tag(module, ec2, count_tag, zone=None):
 
 def _set_none_to_blank(dictionary):
     result = dictionary
-    for k in result.iterkeys():
+    for k in iterkeys(result):
```
Comment (abadger) [inline cloud/amazon/ec2.py]: iterkeys is almost never needed.  Use this instead:

``` python
for k in result:
```

**Pull request merged**